primitives/teerex/src/lib.rs

```diff
@@ -35,31 +35,110 @@ impl Default for SgxBuildMode {
 }
 
 #[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo)]
-pub enum AttestationMethod {
-	Dcap,
+pub enum SgxAttestationMethod {
+	Skip(bool),
 	Ias,
-	Skip,
+	Dcap(bool),
+}
+
+impl Default for SgxAttestationMethod {
+	fn default() -> Self {
+		SgxAttestationMethod::Skip(false)
+	}
+}
+
+const SGX_REPORT_DATA_SIZE: usize = 64;
+#[derive(Debug, Encode, Decode, Copy, Clone, PartialEq, Eq, TypeInfo)]
+#[repr(C)]
+pub struct SgxReportData {
+	pub d: [u8; SGX_REPORT_DATA_SIZE],
+}
+
+impl Default for SgxReportData {
+	fn default() -> Self {
+		SgxReportData { d: [0u8; SGX_REPORT_DATA_SIZE] }
+	}
+}
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo)]
+pub enum SgxStatus {
+	Invalid,
+	Ok,
+	GroupOutOfDate,
+	GroupRevoked,
+	ConfigurationNeeded,
+}
+impl Default for SgxStatus {
+	fn default() -> Self {
+		SgxStatus::Invalid
+	}
 }
 
 #[derive(Encode, Decode, Default, Copy, Clone, PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo)]
-pub struct Enclave<PubKey, Url> {
-	pub pubkey: PubKey, // FIXME: this is redundant information
+pub struct SgxEnclave<Url> {
+	pub report_data: SgxReportData,
 	pub mr_enclave: MrEnclave,
-	// Todo: make timestamp: Moment
-	pub timestamp: u64, // unix epoch in milliseconds
-	pub url: Url,       // utf8 encoded url
-	pub sgx_mode: SgxBuildMode,
+	pub mr_signer: MrSigner,
+	pub timestamp: u64,   // unix epoch in milliseconds
+	pub url: Option<Url>, // utf8 encoded url
+	pub build_mode: SgxBuildMode,
+	pub attestation_method: SgxAttestationMethod,
+	pub status: SgxStatus,
 }
 
-impl<PubKey, Url> Enclave<PubKey, Url> {
+impl<Url> SgxEnclave<Url> {
 	pub fn new(
-		pubkey: PubKey,
+		report_data: SgxReportData,
 		mr_enclave: MrEnclave,
+		mr_signer: MrSigner,
 		timestamp: u64,
-		url: Url,
-		sgx_build_mode: SgxBuildMode,
+		build_mode: SgxBuildMode,
+		status: SgxStatus,
 	) -> Self {
-		Enclave { pubkey, mr_enclave, timestamp, url, sgx_mode: sgx_build_mode }
+		SgxEnclave {
+			report_data,
+			mr_enclave,
+			mr_signer,
+			timestamp,
+			url: None,
+			build_mode,
+			attestation_method: SgxAttestationMethod::default(),
+			status,
+		}
+	}
+
+	pub fn maybe_pubkey<PubKey>(&self) -> Option<PubKey>
```

Comment:
I think this functionality will have to move to MultiEnclave where we can solve this more elegantly I believe
this is an intermediate hack between refactoring steps

Reply:
Yeah, I was also thinking, that this is not ideal, but I did not have any better ideas for now.